src/search.js

```diff
@@ -0,0 +1,132 @@
+}
+
+const showCountryDetails = (country) => {
+
+    const newCurr = (Object.values(country.currencies));
+    const currency = stringFormatter(myJoin(newCurr));
+
+    let language = Object.values(country.languages).toString();
+    language = stringFormatter(language);
+
+    const countryDiv = document.getElementById("country-details");
+
+    const node = document.createElement("DIV");
+    node.className = "country-det-div";
+
+    const imgNode = document.createElement("IMG");
+    imgNode.src = `${country.flags.png}`;
+    imgNode.alt = `${country.demonym} flag`;
+    imgNode.width = 200;
+    imgNode.style.display = "block";
+    imgNode.style.marginLeft = "auto";
+    imgNode.style.marginRight = "auto";
+    imgNode.style.marginBottom = "20px";
+    imgNode.style.boxShadow = "10px 10px 5px rgba(0, 0, 0, 0.5)";
+    imgNode.style.borderRadius = "10px";
+
+    const divider = document.createElement("HR");
+
+    const countryName = document.createElement("H4");
+    const textNode = document.createTextNode(` ${country.name.official.toUpperCase()}`);
+    countryName.appendChild(textNode);
+    countryName.style.marginTop = "15px";
+    countryName.style.textAlign = "center";
+
+    const textNode2 = document.createTextNode(`${country.name.common} is situated in ${country.subregion}. It has a population of ${new Intl.NumberFormat().format(country.population)} people.`);
+    const node2 = document.createElement("BR");
+    const node3 = document.createElement("BR");
+    const textNode3 = document.createTextNode(`The capital is ${country.capital[0]} and you can pay with ${currency}'s.`);
+    const textNode4 = document.createTextNode(`They speak ${language}.`);
+
+    node.appendChild(imgNode);
+    node.appendChild(divider);
+    node.appendChild(countryName);
+    node.appendChild(textNode2);
+    node.appendChild(node2);
+    node.appendChild(textNode3);
+    node.appendChild(node3);
+    node.appendChild(textNode4);
+
+    countryDiv.appendChild(node);
+    countryDiv.style.visibility = "visible";
+    countryDiv.style.opacity = 1;
+    countryDiv.style.display = "block";
+}
+
+const errorMessage = () => {
+    const countryDiv = document.getElementById("country-details");
+    const errorText = document.createTextNode("Invalid Search Term. Try Again!");
+    countryDiv.appendChild(errorText);
+}
+
+const stringFormatter = (string) => {
+    let count = 0;
+    count = (string.match(/,/g) || []).length;
+    switch (count) {
+        case 0: {
+            return string;
+            break;
+        }
+        case 1: {
+            return string.replaceAll(',', " and ");
+            break;
+        }
+        default: {
+            const langs = string.split(",");
+            let newString = "";
+            for (let i = 0; i < count - 1; i++) {
+                newString = langs[i] + ", ";
+            }
+            newString += langs[count - 1] + " and " + langs[count];
+            return newString;
+        }
+    }
```

Comment:
Dit oogt op het eerste gezicht best wel complex. In principe wil je zoiets doen voor een array. Als je dit meegeeft, kun je misschien iets doen als als
```
const [last, ...rest] = [array.pop(), ...array];

return `${rest.length > 0 ? "are": "is"} ${rest.join(", ")} and ${last}`
```